internal/controller/common_nab.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nacv1alpha1 "github.com/migtools/oadp-non-admin/api/v1alpha1"
+	"github.com/migtools/oadp-non-admin/internal/common/constant"
+)
+
+// GetVeleroBackupSpecFromNonAdminBackup gets the BackupSpec from the NonAdminBackup object
+func GetVeleroBackupSpecFromNonAdminBackup(nonAdminBackup *nacv1alpha1.NonAdminBackup) (*velerov1api.BackupSpec, error) {
```

Comment:
duplication of https://github.com/migtools/oadp-non-admin/pull/13/files#diff-de455907ed1c5413fa6248ebea483af559585f8f6277a7c8b370813b9bb254f9R81 ??

Reply:
Right, refactor and I did't remove one.